rfc/src/rfcs/0006-unstable-api.md

```diff
@@ -0,0 +1,98 @@
+- **Feature Name:** Unstable APIs
+- **RFC Tracking Issue**: <https://github.com/model-checking/kani/issues/2279>
+- **RFC PR:** <https://github.com/model-checking/kani/pull/2281>
+- **Status:** Under Review
+- **Version:** 0
+
+-------------------
+
+## Summary
+
+Provide a standard option for users to enable experimental APIs and features in Kani,
+and ensure that those APIs are off by default.
+
+## User Impact
+
+The impact should be minimal, and it is very similar to the current experience when they are trying an experimental feature in Kani.
+ The unstable API will be given a unique identifier, which the user will enable either via command line or via command argument.
+
+## User Experience
+
+Users will have to invoke Kani with:
+```
+--enable-unstable --unstable-feature <feature_identifier>
```

Comment:
Yeah, I agree that this is rather verbose. The problem with `--enable-unstable` today is that it doesn't specify which unstable feature it enables. We could add arguments to it, but it would be a breaking change. We could use `-Z` like the compiler and cargo to specify the actual feature.

Yes, we do need one for each feature identifier. Handling multiple identifiers is not recommended since we accept positional arguments.

Reply:
I like using `-Z`, and it wouldn't be a bad time to remove `--enable-unstable` if we decided to use `-Z<unstable_feature_ID>` for all unstable features.

Reply:
Yes, that'd be great. Another benefit is that we could follow the [same rules they use](https://doc.rust-lang.org/cargo/reference/unstable.html) to require `-Z` for unstable options without an interface.

Reply:
Yes, I was thinking about adding a `[unstable]` table under `kani`.